syntax = "proto3";

// Service for string manipulations
service Capitalizer {
  // Make first letter upper case
  rpc Capitalize (StringRequest) returns (StringResponse) {}

  // Make all letters upper case
  rpc Upper (StringRequest) returns (StringResponse) {}
  
  // Make all letters lower case
  rpc Lower (StringRequest) returns (StringResponse) {}

  // Draw upper case A on image
  rpc DrawA (FileRequest) returns (FileResponse) {}

  // Make all point values upper caes
  rpc UpperPoints (PointsRequest) returns (PointsResponse) {}
}

message StringRequest {
  string message = 1;
}

message StringResponse {
  string message = 1;
}

message FileRequest {
  bytes file = 1;
}

message FileResponse {
  bytes file = 1;
}

message Point {
  sint32 x = 1;
  sint32 y = 2;
  string value = 3;
}

message PointsRequest {
  repeated Point points = 1;
}

message PointsResponse {
  repeated Point points = 1;
}
